gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+                if option not in available_options:
+                    raise subprocess.CalledProcessError(1, command)  
+                else:
+                    pass  
+            else:
+                pass
+        return side_effect
+
+
+    @mock.patch('gslab_scons.configuration_tests.os.path.expanduser')
+    @mock.patch('gslab_scons.configuration_tests.os.path.isdir')
+    def test_check_and_expand_cache_path(self, mock_is_dir, mock_expanduser):
+        mock_expanduser.side_effect = lambda x: re.sub('~', 'Users/lb', x)
+        mock_is_dir.side_effect     = lambda x: x == 'Users/lb/cache'
+
+        configuration_tests.check_and_expand_cache_path('~/cache')
+        configuration_tests.check_and_expand_cache_path('Users/lb/cache')
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_and_expand_cache_path('~/~/cache')
+            configuration_tests.check_and_expand_cache_path('lb/cache')
+            configuration_tests.check_and_expand_cache_path(3)
+
+
+    @mock.patch('gslab_scons.configuration_tests.check_stata_packages')
+    @mock.patch('gslab_scons.configuration_tests.load_yaml_value')
+    @mock.patch('gslab_scons.configuration_tests.get_stata_executable')
+    @mock.patch('gslab_scons.configuration_tests.get_stata_command')
+    def test_check_stata(self, mock_stata_command, mock_stata_exec, 
+                         mock_load_yaml, mock_stata_packages):
+        # Setup
+        def yaml_side_effect(*args, **kwargs):
+            return 'statamp'
+
+        def yaml_side_effect2(*args, **kwargs):
+            return None
+
+        def stata_package_side_effect(*args, **kwargs):
+            command  = args[0]
+            packages = args[1]
+            if command != 'statamp':
+                raise PrerequisiteError()
+            for package in packages:
+                if package != 'yaml':
+                    raise PrerequisiteError()
+
+        mock_load_yaml.side_effect      = yaml_side_effect
+        f = lambda x: x['user_flavor'] if x['user_flavor'] is not None else 'statamp'
+        mock_stata_exec.side_effect     = f
+        mock_stata_command.side_effect  = lambda x: x
+        mock_stata_packages.side_effect = stata_package_side_effect
+
+        # Correct tests
+
+        # Has yaml value
+        self.assertEqual(configuration_tests.check_stata(), 'statamp')
+
+        # No yaml value
+        # Function only returns user-specified executable or none, not the defaults
+        mock_load_yaml.side_effect = yaml_side_effect2
+        self.assertEqual(configuration_tests.check_stata(), None)
+
+        # Failures
+
+        # No yaml value and no default value in path
+        mock_stata_exec.side_effect = lambda x: None
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_stata()
+
+        # Bad package
+        with self.assertRaises(ex_classes.PrerequisiteError):
```

Comment:
I think you should undo line 257 (`mock_stata_exec.side_effect = lambda x: None`) before making this check so we know the error is due to the bad package.